shared/js/background/components/toggle-reports.js

```diff
@@ -0,0 +1,265 @@
+
+        // Note the response and time.
+        await settings.ready()
+        const times = settings.getSetting('toggleReportTimes') || []
+        times.push({ timestamp: Date.now(), accepted })
+        settings.updateSetting('toggleReportTimes', times)
+
+        if (accepted) {
+            await sendBreakageReportForCurrentTab({
+                pixelName: 'protection-toggled-off-breakage-report',
+                reportFlow: 'on_protections_off_dashboard_main'
+            })
+
+            await resolveAfterDelay(2000)
+        }
+
+        // Close the popup UI and reload the page.
+        postPopupMessage({ messageType: 'closePopup' })
+        await reloadCurrentTab()
+    }
+
+    /**
+     * Clear any old recorded response times.
+     *
+     * @returns {Promise<void>}
+     */
+    static async clearExpiredResponses () {
+        const {
+            dismissInterval,
+            promptInterval
+        } = getFeatureSettings('toggleReports')
+
+        const now = Date.now()
+        let dismissCutoff = null
+        let acceptCutoff = null
+
+        if (typeof dismissInterval === 'number') {
+            dismissCutoff = now - (dismissInterval * 1000)
+        }
+        if (typeof promptInterval === 'number') {
+            acceptCutoff = now - (promptInterval * 1000)
+        }
+
+        await settings.ready()
+        let times = settings.getSetting('toggleReportTimes') || []
+        const existingTimesLength = times.length
+        times = times.filter(
+            ({ accepted, timestamp }) =>
+                (accepted && (typeof acceptCutoff !== 'number' ||
+                              timestamp >= acceptCutoff)) ||
+                (!accepted && (typeof dismissCutoff !== 'number' ||
+                               timestamp >= dismissCutoff))
+        )
+
+        if (times.length < existingTimesLength) {
+            settings.updateSetting('toggleReportTimes', times)
+        }
+    }
+
+    /**
+     * Count the number of accepted and declined responses recorded.
+     *
+     * Note: Does not filter away expired times, take care to call
+     *       ToggleReports.clearExpiredResponses() first.
+     *
+     * @returns {Promise<{accepted: number, declined: number}>}
+     */
+    static async countResponses () {
+        await settings.ready()
+        const times = settings.getSetting('toggleReportTimes') || []
+
+        const counts = { accepted: 0, declined: 0 }
+        for (const { accepted } of times) {
+            counts[accepted ? 'accepted' : 'declined']++
+        }
+
+        return counts
+    }
+
+    /**
+     * Check if the toggle report UI flow should been displayed display for the
+     * currently focused tab.
+     *
+     * @param {import('../classes/tab')?} currentTab
+     * @returns {Promise<boolean>}
+     */
+    static async shouldDisplay (currentTab = null) {
+        currentTab = currentTab || (
+            await tabManager.getOrRestoreCurrentTab()
```

Comment:
Why are we not injecting tabManager when creating the component and using `this.tabManager`?

Also, as fas as I can tell, the `currentTab` argument is never provided, so lets remove it.

Reply:
Because this is a static method, which is in turn because it needs to be called from the "setLists" message handler.